src/cclib/io/filewriter.py

```diff
@@ -53,6 +52,10 @@ def __init__(self, ccdata, jobfilename=None, terse=False,
             self.obmol, self.pbmol = self._make_openbabel_from_ccdata()
             self.bond_connectivities = self._make_bond_connectivity_from_openbabel(self.obmol)
 
+        # Check if required attributes are present.
```

Comment:
`filewriter` is the base class. Setting it to an empty list here would clear the values set in derived classes. One way around is to set it in the `__init__` of the derived class, but we had earlier agreed not to do so.